duckduckgo/iOS
# Sanitize location selection against remote locations

Task/Issue URL: https://app.asana.com/0/1203137811378537/1206773442486707/f
BSK: https://github.com/duckduckgo/BrowserServicesKit/pull/764

**Description**:

The VPN geoswitching logic currently doesn’t handle the case that a location becomes unavailable. We should fall back to the country, if a city isn’t available, or nearest if the country isn’t available.

**Steps to test this PR**:
1. Edit the `NetworkProtectionVPNLocationViewModel` to hardcode an extra a location that doesn’t exist and present it on the UI. Try with a country that doesn’t exist as well as a city that doesn’t exist being added to a country that does.
2. Select the location and confirm
3. Check that the fallback logic follow that mentioned in the description.

**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


—
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
